### PR TITLE
Dragon

### PR DESCRIPTION
**Pull request by lemonsqueeze**
Try to deal with virtually connected groups (group has 2 eyes etc)

Needs documenting, haven't put much comments yet.
Some of it is fairly straightforward like dragon_at() dragon_liberties(), but some not esp. group_is_safe()

### CODE REVIEW

**Comment by pasky** [comment]
Depends on PR#48.

This code doesn't seem to be used anywhere (yet)? Also I agree that some documenting would do it good ;)

**Comment by pasky** [comment]
(Also, a better commit message would be nice. I suppose this is a work in progress yet.)

**Comment by lemonsqueeze** [comment]
Added comments to dragon.h

**Comment by pasky** [comment]
I'd appreciate if dragon.h started with a high-level description of what a dragon is and why do we care. :)

It seems we are mixing dragon_*() and group_*() names for dragon.c routines - is that sensible?

I suppose users are in followup commits?

**Comment by lemonsqueeze** [comment]
Something like this ?
I renamed the group_ functions to dragon_  ones, more consistent indeed.
Yes, users come later (ladder and seki code)

File: tactics/dragon.c [hunk 1]
```diff
@@ -0,0 +1,656 @@
...
+	enum stone color = board_at(b, to);
+	if (!g)
+		return 0;
+	
+	foreach_connected_group(b, color, to, dragon_at_handler, &d);
+	return d;
+}
+
+
+#define		GAP_LENGTH        4
```
Comment (pasky) [inline tactics/dragon.c]: So we are looking for 2-stone gap but GAP_LENGTH is 4.  It'd be nice to have a single-sentence comment explaining this...

**Comment by lemonsqueeze** [comment]
More cleanup and unit tests for dragon_is_safe().
Good thing, found a bug i didn't know about while writing it.
I added a comment for the surrounding stones gap code:

    /* Looking for 2-stones horizontal/vertical gap of length GAP_LENGTH extending
     * outwards from lib. For example, something like this:
     *
     *    . X X . . .        X X X X X . . .
     *    . O . X . .        X O O O * * * *
     *    . O * * * *        X O . O * * * *
     *    . O * * * *        X . O . . . . .
     *    . O . X . .      
     */

**Comment by pasky** [comment]
Really impressive work!  I also appreciate your attention to the testsuites a lot.  Thank you.

**Pull request merged**